impeller/renderer/geometry_pass.cc

```diff
@@ -0,0 +1,94 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "impeller/renderer/geometry_pass.h"
+
+#include <iostream>
+#include <tuple>
+
+#include "impeller/core/host_buffer.h"
+#include "impeller/entity/contents/content_context.h"  // nogncheck
+#include "impeller/entity/convex.comp.h"               // nogncheck
+#include "impeller/renderer/render_pass.h"
+
+namespace impeller {
+
+GeometryPass::GeometryPass() {}
+
+constexpr size_t kMaxConvexSegments = 0xFFFFFFFF;  // max uint32_t.
+
+GeometryPass::AccumulatedConvexCommand& GeometryPass::GetOrCreateConvex(
+    size_t count) {
+  FML_DCHECK(count <= kMaxConvexSegments);
+  if (convex_commands_.size() == 0 ||
+      (convex_commands_.size() + count > kMaxConvexSegments)) {
+    convex_commands_.emplace_back(AccumulatedConvexCommand{
+        .count = 0u,
+        .size = 0u,
+        .input_buffer = HostBuffer::Create(),
+        .output_buffer = HostBuffer::Create(),
+        .indirect_buffer = HostBuffer::Create(),
+        .index_buffer = HostBuffer::Create(),
+    });
+  }
+  return convex_commands_.back();
+}
+
+GeometryPassResult GeometryPass::AddPolyline(Path::Polyline polyline,
+                                             const ContentContext& renderer) {
+  using CS = ConvexComputeShader;
+  auto result_size = polyline.points.size() * 3;
+  convex_pipeline_ = renderer.GetConvexComputePipeline();
+  auto& convex_command = GetOrCreateConvex(polyline.points.size());
+
+  // Emplace with no padding as we're going to treat this as a single buffer.
+  std::ignore = convex_command.input_buffer->Emplace(
```

Comment:
This feels sort of hacky, but given that the HostBuffers already sort of act like temporary buffers I didn't want to create another local temporary buffer.